libs/game/sprite.ts

```diff
@@ -673,9 +691,21 @@ class Sprite implements SpriteLike {
 
         const collisionHandlers = game.currentScene().collisionHandlers[other.tileIndex];
         if (collisionHandlers) {
-            collisionHandlers
-                .filter(h => h.kind == this.kind())
-                .forEach(h => h.handler(this));
+            if (!this.ongoingObstacles) this.ongoingObstacles = [];
+            const handlerOccuring = this.ongoingObstacles
+                .some(obs => obs.x === other.x && obs.y === other.y);
+
+            if (!handlerOccuring) {
+                collisionHandlers
+                    .filter(h => h.kind == this.kind())
+                    .forEach(h => {
+                        this.ongoingObstacles.push(other);
+                        control.runInParallel(() => {
```

Comment:
We should avoid runInParallel, it allocates news fibers and messes up scenes

Reply:
Maybe a ``scene.runInParallel``, like ``game.forever``?